Automattic/pocket-casts-ios
# [Paywall Experiments] Current Subscription modal implementation

| 📘 Part of: #1986 
|:---:|

Fixes #2032 

This PR implements our current `PlusPurchaseModal` into the new Paywall container.

![Simulator Screenshot - iPhone 15 Pro - 2024-08-20 at 18 16 53](https://github.com/user-attachments/assets/85fa8a20-9444-4a79-88e9-f0be61857b56)


## To test

> [!NOTE]
> Make sure to enable the `upgradeExperiment` feature flag
> To test this you need to build on a real device with a sandbox user logged in

• CI must be green
• Open the `StoreKit` debug from Xcode -> Debug -> StoreKit -> Manage Transactions

### Entry point 1
• Make sure you are logged out
• Make sure you don't have transactions
• Account tab -> Set Up Account
• Login with a non-Plus user
• You should see the new Paywall Container prompted
• Tap on `Get Pocket Casts Plus` to show the Purchase Modal
• Select a plan and complete the purchase
• After the purchase is completed you should see the Welcome screen

### Entry point 2
• Make sure you are logged out
• Make sure you don't have transactions
• Account tab -> Set Up Account
• Login with a non-Plus user
• You should see the new Paywall Container prompted
• Tap Skip
• Open the settings
• Tap on `Pocket Casts Plus`
• You should see the new Paywall Container prompted
• Tap on `Get Pocket Casts Plus` to show the Purchase Modal
• Select a plan and complete the purchase
• After the purchase is completed you should see the Welcome screen

### Entry point 3
• Make sure you are logged out
• Make sure you don't have transactions
• Account tab
• Open the settings
• Tap on `Pocket Casts Plus`
• You should see the new Paywall Container prompted
• Tap on `Get Pocket Casts Plus`
• Login with a non-Plus user
• You should see the new Paywall Container prompted
• Tap on `Get Pocket Casts Plus` to show the Purchase Modal
• Select a plan and complete the purchase
• After the purchase is completed you should see the Welcome screen


## Checklist
